### PR TITLE
Create snapd-9999.ebuild

### PR DESCRIPTION
**Pull request by kennystrawnmusic**
Need to give people an option to pull straight from git repos

### CODE REVIEW

**Comment by zigford** [comment]
I’ve not created a live ebuild before. Should there be a special keyword for amd64 to block it by default on systems that have accepted ~amd64?

**Comment by kennystrawnmusic** [comment]
Probably not because that raises the risk of creating circular dependencies if they decide to change keywords to “**” to work around it, as I’ve personally experienced. If you look at the contribution, the only difference is just 2 lines.

File: app-emulation/snapd/snapd-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,194 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit bash-completion-r1 golang-base golang-vcs-snapshot linux-info systemd
```
Comment (zigford) [inline app-emulation/snapd/snapd-9999.ebuild:6]: would expect to see git-r3 class inherited or ebuild won't be able to download from a git repo.

File: app-emulation/snapd/snapd-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+DESCRIPTION="Service and tools for management of snap packages"
+HOMEPAGE="http://snapcraft.io/"
+SRC_URI=""
+EGIT_REPO_URI="https://github.com/snapcore/${PN}.git"
+EGIT_BRANCH="master"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="amd64"
```
Comment (zigford) [inline app-emulation/snapd/snapd-9999.ebuild:16]: KEYWORDS should be empty for a live ebuild

File: app-emulation/snapd/snapd-9999.ebuild (hunk 1 above)
Comment (zigford) [reply app-emulation/snapd/snapd-9999.ebuild:6]: I don't think you can use the golang-vcs eclass to build a live snapd as it relies on the entire repo, not just the go packages

**Review by zigford — COMMENTED** [review]
I was not able to get this to install in the sandbox. I'm having a poke around to see what changes would need to be made. Did it work for you?

**Comment by zigford** [comment]
Did you test it? It wouldn't compile for me because go is complaining about missing vendor packages. The tarball version includes them, the git snapshot version does not. I've added a wip live ebuild to the dev branch if you want to take a look at some progress I've made on this.

**Comment by zigford** [comment]
Now updated the ebuild in the dev branch to support live and release versions of snapd. The live version depends on govendor to download the missing vendor modules.

Can you give it a test and see if it meets your needs?

**Comment by kennystrawnmusic** [comment]
Seems to be building fine now. Closing if it’s completely successful.

**Pull request closed without merging**